src/user/controller/password.ts

```diff
@@ -0,0 +1,46 @@
+import Controller from '@curveball/controller';
+import { Context } from '@curveball/core';
+import * as UserService from '../service';
+import { Forbidden, Unauthorized, NotFound } from '@curveball/http-errors';
+import { User } from '../types';
+import * as privilegeService from '../../privilege/service';
+
+class UserPasswordController extends Controller {
+
+  async put(ctx: Context) {
+
+    if (!await privilegeService.hasPrivilege(ctx, 'admin')) {
+      throw new Forbidden('Only users with the "admin" privilege may create new users');
+    }
```

Comment:
from our conversation
`need a new API that allows anyone with admin privileges to set someone's password without knowing the old one`

but I feel that our purpose (NMC) of this endpoint is when the invited user is going through the `onboarding` process, they'll type in their desired password, but the user actually won't have the `admin` access for this.

so maybe removing this privilege check is necessary?
thoughts?

Reply:
No, this check is perfect. We don't want users to change their own passwords.

So you need 'admin' access to do this right now. And one of the things will have admin access, is the nmc api backend. The backend authenticates as itself to do this